File: arrays/fibonacci-number/fibonacci-number.java
class Solution {
  public int fib(int n) {
    if (n == 0) {
      return 0;
    }
    int curr = 1;
    int first = 0;
    int second = 1;

    for (int i = 2; i <= n; i++) {
      curr = first + second;
      first = second;
      second = curr;
    }

    return curr;
  }
}

File: arrays/fibonacci-number/fibonacci-number.go
package fibonaccinumber

func fib(n int) int {
	if n == 0 {
		return 0
	}

	curr := 1
	first := 0
	second := 1

	for i := 2; i <= n; i++ {
		curr = first + second
		first = second
		second = curr
	}
	return curr
}
